### PR TITLE
Resumable Uploads: `application/partial-upload` media type

### PR DESCRIPTION
**Pull request by Acconut**
As discussed in https://github.com/httpwg/http-extensions/issues/2501 and https://github.com/httpwg/http-extensions/issues/2610, the PATCH requests needs a suiting media type to describe the message content.

Since there does not seem to be an existing, suitable media type, [we proposed the `application/partial-upload` media type on the media-types mailing list](https://mailarchive.ietf.org/arch/msg/media-types/XoF-_vBDWWS4KjLFjLlrbnuO8bw/). This PR is a first, rough attempt at including such a media type in the draft.

### CODE REVIEW

File: draft-ietf-httpbis-resumable-upload.md [hunk 1]
```diff
@@ -324,7 +324,7 @@ The client SHOULD NOT automatically retry if a client error status code between
 
 Upload appending is used for resuming an existing upload.
 
-The request MUST use the `PATCH` method and be sent to the upload resource. The `Upload-Offset` field value ({{upload-offset}}) MUST be set to the resumption offset.
+The request MUST use the `PATCH` method and `application/partial-upload` media type and be sent to the upload resource. The `Upload-Offset` field value ({{upload-offset}}) MUST be set to the resumption offset.
```
Comment (smatsson) [inline draft-ietf-httpbis-resumable-upload.md:327]: I feel that this sentence is a bit off, probably due to language barriers on my end. 

Suggested edit:
>  The request MUST use the `PATCH` method with `application/partial-upload` as content type and be sent to the upload resource. The `Upload-Offset` field value ({{upload-offset}}) MUST be set to the resumption offset.

File: draft-ietf-httpbis-resumable-upload.md [hunk 2]
```diff
@@ -336,6 +336,8 @@ The client MUST NOT perform multiple upload transfers for the same upload resour
 
 If the offset indicated by the `Upload-Offset` field value does not match the offset provided by the immediate previous offset retrieval ({{offset-retrieving}}), or the end offset of the immediate previous incomplete successful transfer, the server MUST respond with a `409 (Conflict)` status code.
 
+The server applies the patch document of the `application/partial-upload` media type by appending the unmodified request content to the targeted upload resource.
```
Comment (smatsson) [inline draft-ietf-httpbis-resumable-upload.md:339]: To me "unmodified" reads as "take the entire thing and apply it without changes" which is probably not what is intended here? The server is not allowed to pick bytes that it appends, but I think we need to clarify that the entire request must not complete for the server to append data. I.e. if the connection breaks down after 100 bytes then as much as possible of those 100 bytes _should_ be appended and what is appended _must_ be appended without modification.

File: draft-ietf-httpbis-resumable-upload.md [hunk 3]
```diff
@@ -406,6 +409,10 @@ The `Upload-Complete` request and response header field indicates whether the co
 
 The `Upload-Complete` header field MUST only be used if support by the resource is known to the client ({{feature-detection}}).
 
+# Media Type `application/partial-upload`
+
+The `application/partial-upload` media type describes a contiguous block of data that should be uploaded to a resource. There is no minimum size on the block of data and it might be empty. The start and end of block of data might align with the start and end of the file that should be uploaded, but are not required to do so.
+
```
Comment (smatsson) [inline draft-ietf-httpbis-resumable-upload.md:415]: Minor nit: 

> The start and end of *THE* block of data might align with the start and end of the file that should be uploaded, but are not required to do so.

or 

> The start and end of *THE* data might align with the start and end of the file that should be uploaded, but are not required to do so.


Should we avoid using "file" as the thing to be uploaded and use something more generic like "resource that should be uploaded"?

File: draft-ietf-httpbis-resumable-upload.md (hunk 2 above)
Comment (Acconut) [reply draft-ietf-httpbis-resumable-upload.md:339]: Good idea, thank you.

**Review by guoye-zhang — APPROVED** [review]

**Pull request merged**